src/Microsoft.AspNet.Mvc.Razor/RazorTextWriter.cs

```diff
@@ -181,13 +54,9 @@ public override Task WriteLineAsync()
         /// </summary>
         public override void Flush()
         {
-            if (IsBuffering)
-            {
-                IsBuffering = false;
-                TargetWriter = UnbufferedWriter;
```

Comment:
This used to swtich from buffering to unbuffered when you Flushed. It was hard to keep that behavior with the current flow. That said, I'm not sure if that behavior is useful or expected.